### PR TITLE
Fix blockers for Role UI

### PR DESCRIPTION
**Pull request by MartinTrojans**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

1. Fix delete member in add role modal
2. Disable review enable for admin role
3. Fix Justification for deleting a member

### CODE REVIEW

File: ui/src/components/role/AddRole.js [hunk 1]
```diff
@@ -257,7 +257,10 @@ export default class AddRole extends React.Component {
 
         let role = { name: roleName };
         if (this.state.category === 'regular') {
-            role.roleMembers = this.state.members || [];
+            role.roleMembers =
+                this.state.members.filter(function(member) {
```
Comment (pratikgote) [inline ui/src/components/role/AddRole.js:261]: this.state.members.filter((member) => {})

File: ui/src/components/settings/SettingTable.js [hunk 2]
```diff
@@ -257,6 +257,7 @@ export default class SettingTable extends React.Component {
                 _csrf={this.props._csrf}
                 justificationRequired={this.props.justificationRequired}
                 userProfileLink={this.props.userProfileLink}
+                disabled={role.trim() === 'admin' ? true : false}
```
Comment (pratikgote) [inline ui/src/components/settings/SettingTable.js:260]: we can just have _role.trim() === 'admin'_ no need of ternary operator

**Review by pratikgote — CHANGES_REQUESTED** [review]

File: ui/src/components/role/RoleTable.js [hunk 3]
```diff
@@ -155,6 +155,7 @@ export default class RoleTable extends React.Component {
                     // group rows
                     let roleGroup = (
                         <RoleGroup
+                            key={'group:' + name}
```
Comment (chandrasekhar1996) [inline ui/src/components/role/RoleTable.js:158]: what is this for?

Comment (MartinTrojans) [reply ui/src/components/role/RoleTable.js:158]: To add the unique key id for Role Group element

Comment (MartinTrojans) [reply ui/src/components/role/RoleTable.js:158]: There is a error reporting in the console

**Review by pratikgote — APPROVED** [review]
👍

**Pull request merged**